### PR TITLE
allows for configured additional parameters

### PR DESCRIPTION
**Pull request by mejackreed**
The need for this pull request came from the desire to pass through additional parameters but not wanting to reimplement the controller in a local use.

See https://github.com/sul-dlss/exhibits/pull/970/files

### CODE REVIEW

File: spec/controllers/embed_controller_spec.rb [hunk 1]
```diff
@@ -27,5 +28,11 @@
       json = JSON.parse(response.body)
       expect(json["html"]).to be_blank
     end
+
+    it 'passes along configured additional keys' do
+      allow(OEmbed::Providers).to receive(:get).with('http://some/uri', canvas_index: '5').and_return oembed_obj
+      get :show, params: { url: 'http://some/uri', canvas_index: '5' }
+      expect(response.status).to eq 200
```
Comment (drh-stanford) [inline spec/controllers/embed_controller_spec.rb]: you can use `expect(response).to have_http_status(:ok)` here if you prefer

File: app/controllers/blacklight/oembed/embed_controller.rb [hunk 2]
```diff
@@ -2,18 +2,28 @@ module Blacklight::Oembed
...
         ""
       end
     end
+
+    ##
+    # Allows for blacklight-oembed users to pass through additional configured
+    # parameters
+    def additional_params
+      params
+        .slice(*Blacklight::Oembed::Engine.config.additional_params)
```
Comment (drh-stanford) [inline app/controllers/blacklight/oembed/embed_controller.rb]: So, if I understand this correctly, any parameter keys are allowed in the `additional_params` which is why you're doing `slice.to_unsafe_h`? Does it make sense to restrict which keys are allowed in `config.additional_params`, or is it open-ended?

Comment (mejackreed) [reply app/controllers/blacklight/oembed/embed_controller.rb]: It's open ended here for a consumer to whitelist these parameters as needed.

I can't really see why or what we would restrict to here at the gem level. Open to suggestions though.

File: spec/controllers/embed_controller_spec.rb (hunk 1 above)
Comment (mejackreed) [reply spec/controllers/embed_controller_spec.rb]: Yep! I  was just following the pattern in the tests above for consistency.

**Comment by drh-stanford** [comment]
ok 👍

**Pull request merged**